File: data/CodeLlama/readable_format_HPEC/fortran/file289.f90
subroutine kernel_2mm(alpha, beta, tmp, a, b, c, d,  &
ni, nj, nk, nl)
implicit none

DATA_TYPE, dimension(nj, ni) :: tmp
DATA_TYPE, dimension(nk, ni) :: a
DATA_TYPE, dimension(nj, nk) :: b
DATA_TYPE, dimension(nl, nj) :: c
DATA_TYPE, dimension(nl, ni) :: d
DATA_TYPE :: alpha, beta
integer :: ni, nj, nk, nl
integer :: i, j, k

!$pragma scop
do i = 1, _PB_NI
do j = 1, _PB_NJ
tmp(j,i) = 0.0
do k = 1, _PB_NK
tmp(j,i) = tmp(j,i) + alpha * a(k,i) * b(j,k)
end do
end do
end do

do i = 1, _PB_NI
do j = 1, _PB_NL
d(j,i) = d(j,i) * beta
do k = 1, _PB_NJ
d(j,i) = d(j,i) + tmp(k,i) * c(j,k)
end do
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file289.cpp
void kernel_2mm(DATA_TYPE alpha, DATA_TYPE beta, DATA_TYPE tmp[nj][ni], DATA_TYPE a[nk][ni], DATA_TYPE b[nj][nk], DATA_TYPE c[nl][nj], DATA_TYPE d[nl][ni], int ni, int nj, int nk, int nl)
{
    int i, j, k;
#pragma scop
    for(i = 0; i < _PB_NI; i++){
        for(j = 0; j < _PB_NJ; j++){
            tmp[j][i] = 0.0;
            for(k = 0; k < _PB_NK; k++){
                tmp[j][i] = tmp[j][i] + alpha * a[k][i] * b[j][k];
            }
        }
    }
    for(i = 0; i < _PB_NI; i++){
        for(j = 0; j < _PB_NL; j++){
            d[j][i] = d[j][i] * beta;
            for(k = 0; k < _PB_NJ; k++){
                d[j][i] = d[j][i] + tmp[k][i] * c[j][k];
            }
        }
    }
#pragma endscop
}
